Improve C++ legacy crypto algorithm query

gsutherland-trailofbits opened the pull request:
As it stands the query outputs detections that look like this:

> Potential use of legacy cryptographic algorithm cast_to_string detected

This PR alters the message to include the detected cipher name, to make it easier to identify which substring triggered the detection:

> Potential use of legacy cryptographic algorithm CAST detected in function name cast_to_string

This makes things a bit less confusing and makes it easier to quickly triage false positives.

This PR also reduces false positive detections for the DES cipher by using a regex match to avoid common words like "description" and "nodes". The regex has been tested to ensure that fragments such as "descrypt", "3des", "tripledes", etc. are still properly detected.

gsutherland-trailofbits committed:
Improve C++ legacy crypto algorithm query

Adds the name of the detected algorithm to help identify the substring that triggered the detection, and reduces false positives when detecting DES by using a regex to exclude common words like "description" and "nodes".

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/trailofbits/codeql-queries?pullRequest=22) <br/>All committers have signed the CLA.

gsutherland-trailofbits committed:
Fix editing error in legacy cryptography algorithm query

The "DES" string is covered by the regex and should not be in the first list. This was a typo made during testing.

GrosQuildu reviewed (dismissed):
Thanks!

GrosQuildu committed:
Merge branch 'main' into gsutherland-trailofbits-cpp-legacy-algorithm
